src/Sylius/Bundle/ApiBundle/Form/Type/ProductVariantType.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ApiBundle\Form\Type;
+
+use Sylius\Bundle\ProductBundle\Form\Type\ProductVariantType as BaseProductVariantType;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+final class ProductVariantType extends AbstractType
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder->addEventListener(FormEvents::PRE_SUBMIT , function (FormEvent $event) {
+            $data = $event->getData();
+            $form = $event->getForm();
+
+            if (!array_key_exists('onHand', $data)) {
+                $data['onHand'] = $form->getNormData()->getOnHand();
```

Comment:
We could simply remove the form field from the form if onHand is not sent? WDYT?